storage/src/mmr/hasher.rs

```diff
@@ -20,17 +20,19 @@ pub trait Hasher<H: CHasher>: Send + Sync {
 
     /// Computes the root for an MMR given its size and an iterator over the digests of its peaks in
     /// decreasing order of height.
-    fn root_digest<'b>(
+    fn root_digest<'a>(
         &mut self,
         size: u64,
-        peak_digests: impl Iterator<Item = &'b H::Digest>,
+        peak_digests: impl Iterator<Item = &'a H::Digest>,
     ) -> H::Digest;
 
     /// Compute the digest of a byte slice.
     fn digest(&mut self, data: &[u8]) -> H::Digest;
 
     /// Access the inner [CHasher] hasher.
     fn inner(&mut self) -> &mut H;
+
+    fn clone(&self) -> impl Hasher<H>;
```

Comment:
I think this would be better represented by:

```
pub trait Hasher<H: CHasher>: Send + Sync + Clone
```

Reply:
Changed to "duplicate()"

Reply:
I suggest `fork` (as it borrows some processor-y lingo for what it is).